src/terrain/index.js

```diff
@@ -0,0 +1,157 @@
+const { getRandomTerrainFunction } = require('./terrainFunctions');
+const terrainGenerator = (boardBounds = { x: 100, y: 100 }, terrainBounds = { x: 5, y: 5 }, numberOfTerrainElements = 5) => {
+    let { mockArray, mockWidth, mockHeight } = createMockArray(boardBounds, terrainBounds);
+    const keyPoint = {
+        x: Math.floor(Math.random() * mockWidth),
+        y: Math.floor(Math.random() * mockHeight)
+    };
+    mockArray = generateTerrain(mockArray, keyPoint, numberOfTerrainElements, mockWidth, mockHeight);
+    return mockArray;
+};
+
+createMockArray = (boardBounds, terrainBounds) => {
+    const boardX = boardBounds.x;
+    const boardY = boardBounds.y;
+    const terrainX = terrainBounds.x;
+    const terrainY = terrainBounds.y;
+
+    const mockArray = [];
+    const mockWidth =  boardX/terrainX;
+    const mockHeight = boardY/terrainY;
+
+    const defaultInfo = {
+        filled: false,
+        north: true,
+        south: true,
+        east: true,
+        west: true
+    };
+
+    for(let i = 0; i < mockWidth; i++) {
+        mockArray[i] = [];
+        for (let j = 0; j < mockHeight; j++) {
+            const info = { ...defaultInfo };
+            if (i === 0) {
+                info.west = false;
+            } else if (i === mockWidth - 1) {
+                info.east = false;
+            }
+            if (j === 0) {
+                info.north = false;
+            } else if (j === mockHeight - 1) {
+                info.south = false;
+            }
+            mockArray[i].push(info);
+        }
+    }
+    return {
+        mockArray,
+        mockWidth,
+        mockHeight
+    };
+};
+
+const deepClone = (toCopy) => {
+    const toReturn = [];
+    toCopy.forEach((row, rowNum) => {
+        toReturn[rowNum] = [];
+        row.forEach(elem => {
+            toReturn[rowNum].push({ ...elem });
+        });
+    });
+    return toReturn;
+};
+
+
+const cleanUpBoard = (board) => {
```

Comment:
I'm not quite sure what this is for. It looks like it's iterating through the board and reassigning every value

Reply:
yeah maybe its not quite clear from the code. The idea behind the algorithm is:

let keyPoint be an available point on the board
if all available points can reach the keyPoint
then there exists a pathway to any available point from any available point

where availability is defined to be that point is inside the bounds of the array, and the point is not "filled" with terrain